pkg/reconciler/kubernetes/tektonpruner/reconcile.go

```diff
@@ -0,0 +1,163 @@
+
+// ReconcileKind compares the actual state with the desired, and attempts to
+// converge the two.
+func (r *Reconciler) ReconcileKind(ctx context.Context, tp *v1alpha1.TektonPruner) pkgreconciler.Event {
+	logger := logging.FromContext(ctx).With("name", tp.GetName())
+	tp.Status.InitializeConditions()
+	tp.Status.SetVersion(r.prunerVersion)
+
+	if tp.GetName() != v1alpha1.TektonPrunerResourceName {
+		msg := fmt.Sprintf("Resource ignored, Expected Name: %s, Got Name: %s",
+			v1alpha1.TektonPrunerResourceName,
+			tp.GetName(),
+		)
+		logger.Error(msg)
+		tp.Status.MarkNotReady(msg)
+		return nil
+	}
+
+	if err := r.targetNamespaceCheck(ctx, tp); err != nil {
+		return err
+	}
+
+	//Make sure TektonPipeline is installed before proceeding with
+	//TektonPruner
+	if _, err := common.PipelineReady(r.pipelineInformer); err != nil {
+		if err.Error() == common.PipelineNotReady || err == v1alpha1.DEPENDENCY_UPGRADE_PENDING_ERR {
+			tp.Status.MarkDependencyInstalling("tekton-pipelines is still installing")
+			// wait for pipeline status to change
+			return v1alpha1.REQUEUE_EVENT_AFTER
+		}
+		// (tektonpipeline.operator.tekton.dev instance not available yet)
+		tp.Status.MarkDependencyMissing("tekton-pipelines does not exist")
+		return err
+	}
+	tp.Status.MarkDependenciesInstalled()
+
+	// Pass the object through defaulting
+	tp.SetDefaults(ctx)
+
+	if err := r.installerSetClient.RemoveObsoleteSets(ctx); err != nil {
+		logger.Error("failed to remove obsolete installer sets: %v", err)
+		return err
+	}
+
+	if err := r.extension.PreReconcile(ctx, tp); err != nil {
+		msg := fmt.Sprintf("PreReconciliation failed: %s", err.Error())
+		logger.Error(msg)
+		if err == v1alpha1.REQUEUE_EVENT_AFTER {
+			return err
+		}
+		tp.Status.MarkPreReconcilerFailed(msg)
+		return nil
+	}
+
+	//Mark PreReconcile Complete
+	tp.Status.MarkPreReconcilerComplete()
+
+	// Ensure webhook deadlock prevention before applying the manifest
+	//if err := common.PreemptDeadlock(ctx, &r.manifest, r.kubeClientSet, v1alpha1.TektonPrunerResourceName); err != nil {
+	//	msg := fmt.Sprintf("Failed to preempt webhook deadlock: %s", err.Error())
+	//	logger.Error(msg)
+	//	return err
+	//}
+
+	if err := r.installerSetClient.MainSet(ctx, tp, &r.manifest, filterAndTransform(r.extension)); err != nil {
+		msg := fmt.Sprintf("Main Reconcilation failed: %s", err.Error())
+		logger.Error(msg)
+		if err == v1alpha1.REQUEUE_EVENT_AFTER {
+			return err
+		}
+		tp.Status.MarkInstallerSetNotReady(msg)
+		return nil
+	}
+
+	if err := r.extension.PostReconcile(ctx, tp); err != nil {
+		msg := fmt.Sprintf("PostReconciliation failed: %s", err.Error())
+		logger.Error(msg)
+		if err == v1alpha1.REQUEUE_EVENT_AFTER {
+			return err
+		}
+		tp.Status.MarkPostReconcilerFailed(msg)
+		return nil
+	}
+
+	// Mark PostReconcile Complete
+	tp.Status.MarkPostReconcilerComplete()
+	return nil
+}
+
+func (r *Reconciler) targetNamespaceCheck(ctx context.Context, comp v1alpha1.TektonComponent) error {
```

Comment:
Instead of this function, can we reuse this function https://github.com/tektoncd/operator/blob/main/pkg/reconciler/common/targetnamespace.go#L35